render/egl.c

```diff
@@ -365,6 +359,53 @@ EGLImage wlr_egl_create_image_from_dmabuf(struct wlr_egl *egl,
 		attribs[atti++] = EGL_DMA_BUF_PLANE0_MODIFIER_HI_EXT;
 		attribs[atti++] = attributes->modifier[0] >> 32;
 	}
+
+	if (attributes->n_planes > 1) {
+		attribs[atti++] = EGL_DMA_BUF_PLANE1_FD_EXT;
+		attribs[atti++] = attributes->fd[1];
+		attribs[atti++] = EGL_DMA_BUF_PLANE1_OFFSET_EXT;
+		attribs[atti++] = attributes->offset[1];
+		attribs[atti++] = EGL_DMA_BUF_PLANE1_PITCH_EXT;
+		attribs[atti++] = attributes->stride[1];
+		if (has_modifier) {
+			attribs[atti++] = EGL_DMA_BUF_PLANE1_MODIFIER_LO_EXT;
+			attribs[atti++] = attributes->modifier[1] & 0xFFFFFFFF;
+			attribs[atti++] = EGL_DMA_BUF_PLANE1_MODIFIER_HI_EXT;
+			attribs[atti++] = attributes->modifier[1] >> 32;
+		}
+	}
+
+	if (attributes->n_planes > 2) {
+		attribs[atti++] = EGL_DMA_BUF_PLANE2_FD_EXT;
+		attribs[atti++] = attributes->fd[2];
+		attribs[atti++] = EGL_DMA_BUF_PLANE2_OFFSET_EXT;
+		attribs[atti++] = attributes->offset[2];
+		attribs[atti++] = EGL_DMA_BUF_PLANE2_PITCH_EXT;
+		attribs[atti++] = attributes->stride[2];
+		if (has_modifier) {
+			attribs[atti++] = EGL_DMA_BUF_PLANE2_MODIFIER_LO_EXT;
+			attribs[atti++] = attributes->modifier[2] & 0xFFFFFFFF;
+			attribs[atti++] = EGL_DMA_BUF_PLANE2_MODIFIER_HI_EXT;
+			attribs[atti++] = attributes->modifier[2] >> 32;
+		}
+	}
+
+	if (attributes->n_planes > 3) {
+		attribs[atti++] = EGL_DMA_BUF_PLANE3_FD_EXT;
+		attribs[atti++] = attributes->fd[3];
+		attribs[atti++] = EGL_DMA_BUF_PLANE3_OFFSET_EXT;
+		attribs[atti++] = attributes->offset[3];
+		attribs[atti++] = EGL_DMA_BUF_PLANE3_PITCH_EXT;
+		attribs[atti++] = attributes->stride[3];
+		if (has_modifier) {
+			attribs[atti++] = EGL_DMA_BUF_PLANE3_MODIFIER_LO_EXT;
+			attribs[atti++] = attributes->modifier[3] & 0xFFFFFFFF;
+			attribs[atti++] = EGL_DMA_BUF_PLANE3_MODIFIER_HI_EXT;
+			attribs[atti++] = attributes->modifier[3] >> 32;
+		}
+	}
+
+
 	attribs[atti++] = EGL_NONE;
```

Comment:
Maybe adding an assertion here would be safer:

```c
assert(atti < sizeof(attribs)/sizeof(attribs[0]));
```

Reply:
Since the size of attribs is fixed we're safe but in case we change the number of planes we might forget to increase it so I added the assert.